buildkite/resource_team_member_test.go

```diff
@@ -0,0 +1,238 @@
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MEMBER"),
+				),
+			},
+			{
+				Config: testAccTeamMemberConfigBasic("MAINTAINER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckTeamMemberRemoteValues(&resourceTeamMember, "MAINTAINER"),
+				),
+			},
+		},
+	})
+}
+
+// Confirm that this resource can be imported
+func TestAccTeamMember_import(t *testing.T) {
+	var resourceTeamMember TeamMemberNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTeamMemberResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamMemberConfigBasic("MEMBER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &resourceTeamMember),
+					// Confirm the team has the correct values in Buildkite's system
+					resource.TestCheckResourceAttr("buildkite_team_member.test", "role", "MEMBER"),
+				),
+			},
+			{
+				// re-import the resource and confirm they match
+				ResourceName:      "buildkite_team_member.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+// Confirm that this resource can be removed
+func TestAccTeamMember_disappears(t *testing.T) {
+	var teamMember TeamMemberNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckTeamMemberResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamMemberConfigBasic("MEMBER"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team member exists in the buildkite API
+					testAccChecKTeamMemberExists("buildkite_team_member.test", &teamMember),
+					// Ensure its removed
+					testAccCheckResourceDisappears(testAccProvider, resourceTeamMember(), "buildkite_team_member.test"),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccTeamMemberConfigBasic(role string) string {
+	config := `
+		resource "buildkite_team" "test" {
+			name = "acceptance testing"
+			description = "a cool team for testing"
+			privacy = "VISIBLE"
+			default_team = true
+			default_member_role = "MEMBER"
+		}
+		resource "buildkite_team_member" "test" {
+		    role = "%s"
+			team_id = buildkite_team.test.id
+			user_id = "VXNlci0tLWRlOTdmMjBiLWJkZmMtNGNjOC1hOTcwLTY1ODNiZTk2ZGEyYQ=="
```

Comment:
@yob @jeremybumsted I could do with some ideas on how to handle this.
This ID is my email used for these acceptance tests which is a bit flaky. Should we create a machine account for this or do you have other ideas?

Reply:
How would a user of the provider find the correct user graphql ID to include here?

Maybe if we know that, we can find a way to replicate it in the acceptance tests?

If the answer is "look in up on buildkite.com" somewhere, that probably won't help in the tests. Would it be possible to add a data source for looking up a users graphql ID?

Reply:
Yeah I think a datasource would be good but should be followed up in another PR. I'll create a ticket for that

Reply:
yer, staying focused in this PR is wise :+1:

Reply:
What should we do for the integration tests here until then do you think? Use a real person a machine user? I think even if we waited for a datasource, we would still need one of these types of users added to the organization in order for the test to work.

Reply:
How does one get GraphQL IDs of users in their organization? The example GraphQL query given in the docs can extract an existing team membership GraphQL ID for the purpose of importing existing team members into a team resource, but the docs also note that this is different than the user's GraphQL ID, and I can't find any way to get the latter.

> Yeah I think a datasource would be good but should be followed up in another PR. I'll create a ticket for that

Did you ever create that ticket?

Reply:
Answering myself here, this did the trick for us: https://buildkite.com/docs/apis/graphql/graphql-cookbook#search-for-an-organization-member-by-email-address